### DISCUSSION TITLE
Transport with SHM and UDP simultaneously

### DISCUSSION THREAD

**Discussion by dasisttao** [eProsima/Fast-DDS#3478] [Q&A]
Hello, is it possible to set the transport protoocol SHM and UDP simultaneously? My intent was making a publisher with these 2 possibilities so that a subscriber in the same host can communicate with the publisher using SHM and another subscriber in thre remote can communicate with this same publisher using UDP.

**Comment by MiguelCompany** [comment]
That is indeed the default behavior

**Reply by dasisttao** [reply]
Is it possible to configure that explicitly using xml? When I try `<type>UDPv4 | SHM</type> ` or just `<type>UDPv4</type> <type>SHM</type>`it seems no working.

**Reply by JLBuenoLopez** [reply]
Hi @dasisttao,

You need to define two different `<transport_descriptors>`, each one with its own identifier and the corresponding type. And the, you should add both transports into the `<userTransports>` defined within the DomainParticipant profile.

```
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <transport_descriptors>
        <transport_descriptor>
            <transport_id>udp_transport</transport_id>
            <type>UDPv4</type>
        </transport_descriptor>
        <!-- Create another descriptor for the SHM transport -->
        <transport_descriptor>
            <transport_id>shm_transport</transport_id>
            <type>SHM</type>
        </transport_descriptor>
    </transport_descriptors>

    <participant profile_name="Participant">
        <rtps>
            <!-- Link the Transport Layers to the Participant -->
            <userTransports>
                <transport_id>udp_transport</transport_id>
                <transport_id>shm_transport</transport_id>
            </userTransports>
            <useBuiltinTransports>false</useBuiltinTransports>
        </rtps>
    </participant>
</profiles>
```
